### PR TITLE
BIT-82 Bump Koin version to 1.0.1

### PR DESCRIPTION
**Pull request by mmoczkowski**

### CODE REVIEW

**Comment by maciej-wichrowski** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/ui/build/ui/BuildFragment.kt:42: Check whether you really need !! or if you can replace it by ? or similar.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/ui/build/ui/BuildFragment.kt:50: Check whether you really need !! or if you can replace it by ? or similar.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/java/io/stanwood/bitrise/ui/artifacts/ui/ArtifactsFragment.kt [hunk 1]
```diff
@@ -56,15 +56,16 @@ class ArtifactsFragment: Fragment() {
 
     private val viewModel: ArtifactsViewModel
         by inject(parameters = {
-            mapOf(
-                    Properties.TOKEN to token,
-                    Properties.APP to app,
-                    Properties.BUILD to build)
+            parametersOf(
+                token,
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/artifacts/ui/ArtifactsFragment.kt]: So now those are not named any more?

File: app/src/main/java/io/stanwood/bitrise/ui/build/di/BuildModule.kt [hunk 2]
```diff
@@ -22,38 +22,43 @@
...
+import org.koin.dsl.module.module
 
-val buildModule = applicationContext {
+val buildModule = module {
 
     /**
      * View Model
      */
-    factory { args ->
+    factory { (fragmentManager: FragmentManager, token: String, build: Build, app: App) ->
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/build/di/BuildModule.kt]: Nice, much better than the properties! 😍

File: app/src/main/java/io/stanwood/bitrise/ui/build/ui/BuildFragment.kt [hunk 3]
```diff
@@ -32,16 +32,18 @@ import io.stanwood.bitrise.databinding.FragmentBuildBinding
...
 
     private val viewModel: BuildViewModel
         by inject(parameters = {
-            mapOf(
-                Properties.TOKEN to token,
-                Properties.BUILD to build,
-                Properties.APP to app,
-                Properties.FRAGMENT_MANAGER to fragmentManager!!)
+            parametersOf(
+                fragmentManager!!,
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/build/ui/BuildFragment.kt]: Can we somehow get rid of the `!!`?

File: app/src/main/java/io/stanwood/bitrise/util/databinding/ObservableViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import android.arch.lifecycle.ViewModel
+import android.databinding.Observable
+import android.databinding.PropertyChangeRegistry
+
+/**
+ * A ViewModel that is also an Observable,
+ * to be used with the Data Binding Library.
+ */
+open class ObservableViewModel : ViewModel(), Observable {
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/util/databinding/ObservableViewModel.kt]: Is this actually being used?

**Pull request merged**